hudi-sync/hudi-hive-sync/src/main/java/org/apache/hudi/hive/util/HiveSchemaUtil.java

```diff
@@ -65,6 +67,34 @@ public class HiveSchemaUtil {
   public static final String BINARY_TYPE_NAME = "binary";
   public static final String DATE_TYPE_NAME = "date";
 
+  private static final String DATABASE_NAME = "DATABASE_NAME";
+  private static final String EXTERNAL = "external";
+  private static final String TABLE_NAME = "TABLE_NAME";
+  private static final String LIST_COLUMNS = "columns";
+  private static final String PARTITIONS = "partitions";
+  private static final String ROW_FORMAT = "row_format";
+  private static final String LOCATION = "location";
+  private static final String LOCATION_BLOCK = "location_block";
+  private static final String PROPERTIES = "properties";
+  private static final String BUCKETS = "buckets";
+
+  private static final String CREATE_DATABASE_STMT =
+      "CREATE DATABASE IF NOT EXISTS <" + DATABASE_NAME + ">";
+
+  private static final String CREATE_TABLE_TEMPLATE =
+      "CREATE <" + EXTERNAL + ">TABLE <if(" + DATABASE_NAME + ")>`<" + DATABASE_NAME + ">`.<endif>"
+      + "`<" + TABLE_NAME + ">`(\n"
+      + "<" + LIST_COLUMNS + ">)\n"
+      + "<" + PARTITIONS + ">\n"
+      + "<" + BUCKETS + ">\n"
+      + "<" + ROW_FORMAT + ">\n"
+      + "<" + LOCATION_BLOCK + ">"
+      + "TBLPROPERTIES (\n"
+      + "<" + PROPERTIES + ">)";
```

Comment:
two concerns here:
1. Where does the antlr jar comes from? Do we need to package antlr jar explicitly
2. the antlr grammar template is not that straight-forward, should refactoring like using the stringBuilder with params be enough now?

Reply:
@danny0405 Thank you very much for helping review the code!

1. Antlr was introduced into the Hudi project in [HUDI-4111] Bump ANTLR runtime version in Spark 3.x (#5606), we can find the reference of antlr in the parent pom.xml
```
<antlr.version>4.8</antlr.version>
```
`hudi-hive-sync` references `hive-exec`, so we can use `antlr` directly.

2. Thank you for your question! I agree with your point of view, in some scenarios, the antlr grammar template is not that straight-forward. But for generating sql, I think we can use it, because the template can better describe the components of sql.

HiveSchemaUtil#CREATE_TABLE_TEMPLATE
```
private static final String CREATE_TABLE_TEMPLATE =
      "CREATE <" + EXTERNAL + ">TABLE <if(" + DATABASE_NAME + ")>`<" + DATABASE_NAME + ">`.<endif>"
      + "`<" + TABLE_NAME + ">`(\n"
      + "<" + LIST_COLUMNS + ">)\n"
      + "<" + PARTITIONS + ">\n"
      + "<" + BUCKETS + ">\n"
      + "<" + ROW_FORMAT + ">\n"
      + "<" + LOCATION_BLOCK + ">"
      + "TBLPROPERTIES (\n"
      + "<" + PROPERTIES + ">)";
```

Through this template, we can know the elements of the sql to be generated, such as the name of the table, the name of the database, and various attributes of the table. It is easier to read this part of the code.

Reply:
Not a fan of antlr, this introduces unnecessary complexity and dependency of the antlr jar, how about we refactor the code using just a string builder, each sub-clause can be built separately.

Reply:
I agree with you. I will refactor this part of code with string builder.

Reply:
@danny0405 Thank you very much for helping to review the code and providing suggestions for improvement! I have refactored this part of the code using StringBuilder, as we discussed, to improve its readability. I also added some comments to provide additional context and clarification. Sorry for the delayed response.

When you have time, could you please review this PR again? Thank you very much!